roles/cifmw_cephadm/templates/ceph_rgw.yml.j2

```diff
@@ -13,6 +13,11 @@ spec:
   rgw_frontend_port: 8082
   rgw_realm: default
   rgw_zone: default
+{% if rgw_frontend_cert is defined %}
```

Comment:
by default this is skipped if we don't pass/set a path to `cifmw_cephadm_certificate` || `cifmw_cephadm_key`

Reply:
I'd double-ensure things by checking it's not empty either. I remember seeing some weird things with `set_fact` associated to a `when` (basically, the `block` instructs ansible to apply the same when/vars/environment to each of task nested in that block - it's evaluating them anyway).

Reply:
ack, @katarimanojk do you want to test the same by having also `len(rgw_frontend_cert) > 0` ? This way we can avoid the situation described by Cedric.

Reply:
@fmount sure , i will consider the comment from cedric , test it and make changes accordingly.

Reply:
just made a quick test locally:
```YAML
- hosts: localhost
  gather_facts: false
  tasks:
    - name: Maybe set fact
      when:
        - false
      ansible.builtin.set_fact:
        foo: 'bar'

    - name: Debug foo fact
      ansible.builtin.debug:
        var: foo
```

seems to work as it should:
```
TASK [Maybe set fact] *
skipping: [localhost]

TASK [Debug foo fact] *
ok: [localhost] => {
    "foo": "VARIABLE IS NOT DEFINED!"
}
```

same if we nest the task in a block. So my memory probably deserved me on that one.

Reply:
(so., solving this one :) )